bootstrap/kubeadm/types/upstreamv1beta1/types.go

```diff
@@ -220,7 +220,7 @@ type NodeRegistrationOptions struct {
 
 	// Taints specifies the taints the Node API object should be registered with. If this field is unset, i.e. nil, in the `kubeadm init` process
 	// it will be defaulted to []v1.Taint{'node-role.kubernetes.io/master=""'}. If you don't want to taint your control-plane node, set this field to an
-	// empty slice, i.e. `taints: {}` in the YAML file. This field is solely used for Node registration.
+	// empty slice, i.e. `taints: []` in the YAML file. This field is solely used for Node registration.
```

Comment:
I was not sure if it is okay to change the upstream kubeadm types, but without changing this one description in the generated CRDs still contained the `taints: {}` example.

Reply:
Let's keep the upstream types as is. We generally try to keep the diffs there minimal and the godoc here is not exposed anywhere to CAPI users.

Reply:
This godoc is exposed directly to CAPI users for v1alpha3: https://github.com/kubernetes-sigs/cluster-api/blob/8b86290c254012ed2a828f8bdcd6dfa0dc02d0c1/bootstrap/kubeadm/api/v1alpha3/kubeadmconfig_types.go#L48

But as @killianmuldoon said, we should probably skip making these changes to out-of-support CRD versions, so I am going to remove this change from the PR.

Reply:
Ah good point, you're right.